lib/utils/types.ts

```diff
@@ -12,27 +11,21 @@ export type Tagged<T, S extends t.mixed, A> = t.Type<A & T, S>;
  * Tags an io-ts type with an interface T
  */
 export const tag = <T>() => <S, A>(type: t.Type<A, S>): Tagged<T, S, A> =>
+  // tslint:disable-next-line:no-any
   type as any;
 
-const getObjectValues = <T extends object>(obj: T): ReadonlyArray<string> =>
-  Object.keys(obj).reduce<ReadonlyArray<string>>(
-    (acc, key) => [...acc, (obj as any)[key]],
-    []
-  );
-
 /**
  * Creates an io-ts Type from an enum
  */
-export const enumType = <E>(e: {}, name: string): t.Type<any, E> => {
-  const values = getObjectValues(e);
-  const isE: (v: any) => boolean = v =>
-    typeof v === "string" && values.indexOf(v) >= 0;
-  return new t.Type<any, E>(
-    name,
-    (v): v is E => isE(v),
-    (v, c) => (isE(v) ? t.success(v) : t.failure(v, c)),
-    t.identity
+export const enumType = <E>(e: object, name: string): t.Type<E> => {
+  // get object values as keys
+  const keys = Object.keys(e).reduce(
```

Comment:
not sure I understand why this is needed, apparently you can just pass `e` to `t.keyof` instead of `keys`?

Reply:
only if you assume that keys and values are always the same (which it was not our assumption even in the current configuration, which is why we need a call to getObjectValues). probably it is more correct to work on values as keys can be integers (aka, key values depend on typescript internals while values are more trustworthy)

Reply:
yes indeed!

Reply:
that's the reason the method collects all the values and ignore keys (like the old getObjectValues)

Reply:
sorry I didn't understand at first, now I see what is going ok, perhaps it's worth splitting the method inside the reduce and/or add some comments